Add information pages to project instructions

ofr1tz opened the pull request:
Information pages are parts of MapSwipe tutorials, consisting in titled pages that can have text and image blocks as content. This adds a new ProjectInfo component that displays information pages and the already existent project type specific user instructions. The project information dialog is opened by default if a user selects a project they have not yet contributed to, but can also be reached through a button in the project header toolbar.

Refers #13

Hagellach37 commented:
some minor comments:
* how do we want to handle the [left] [right] hot keys? They do not always work for me. User might not notice that they have moved the map view in the background by going right with the hot keys. So should we disable it when the instructions are shown? this just happens when first opening the project

not sure how we want to handle things like this. The text suggests that there will be more examples, but actually we will not provide them.
![image](https://github.com/user-attachments/assets/d0d04d87-ba90-4b98-aa34-cf76c05f3113)

* however for most projects I've checked it looked good

We might want to improve the visual style, e.g. for wide screens
![image](https://github.com/user-attachments/assets/7354693b-3fb2-4785-a7bf-331dd85c642e)

Hagellach37 commented on src/components/CompareProject.vue:
```diff
@@ -111,6 +118,10 @@ export default defineComponent({
     },
+    createInformationPages,
+    createFallbackInformationPages() {
+      return undefined
```
this just means that there are no fallback information pages defined for this project type? It might be worth adding this as as comment here.

ofr1tz replied:
There are no fallback information pages for Compare projects in the app, so neither should we in the web app. I've added a comment here.

Hagellach37 commented on src/components/DigitizeProject.vue:
```diff
@@ -86,6 +93,10 @@ export default defineComponent({
     },
+    createInformationPages,
+    createFallbackInformationPages() {
+      return undefined
```
same as above: why is this undefined?

ofr1tz replied:
Added a comment here as well. Tutorials for Digitize projects do not exist at the moment. Fallback information pages can better be discussed at the time we implement tutorials for this project type.

Hagellach37 commented on src/components/DigitizeProjectInstructions.vue:
```diff
@@ -7,126 +7,85 @@ export default defineComponent({
+        <v-col>{{ $t('digitizeProjectInstructions.deleteFeature') }}</v-col>
+      </v-row>
+    </div>
+    <div class="text-h6 mt-10">{{ $t('findProjectInstructions.useButtonsToNavigate') }}</div>
```
I'm confused why it says here `findProjectInstructions.useButtonsToNavigate` -> why findProject, as we are in the DigitizeProject. It happens a few more times in this file. Do we need to move these translations to another part as well? e.g. just ProjectInstructions?

ofr1tz replied:
Agreed, I restructured project instructions translations.

Hagellach37 commented on src/components/FindProject.vue:
```diff
@@ -22,6 +25,7 @@ export default defineComponent({
       type: Object,
       require: true,
     },
+    // TODO: change this so that instructions dialog is opened with tutorial, but not with project
```
is this still the case?

Hagellach37 commented on src/components/FindProjectInstructions.vue:
```diff
@@ -23,184 +23,144 @@ export default defineComponent({
-        </div>
+      </div>
+    </span>
+    <div class="text-h6 mt-10">{{ $t('compareProjectInstructions.whereIamTitle') }}</div>
```
here again it's confusing that the translation comes from the compareProject

Hagellach37 reviewed (commented):
Hey @ofr1tz ,
thanks for implementing this cool feature. It's great that we can now display the instruction pages for all project types.

The only thing I might adjust is how we handle the translations. Would it make sense to just have entries refering to "projectInstructions" instead of referring sometimes to "findProjectInstructions", "validateProjectInstructions", ...?

What's your opinion on this?

ofr1tz commented:
Hi @Hagellach37 , I addressed your comments with the last commits:

* Restructured translations
* Deactivated arrow shortkeys for project navigation while the dialog is open
* Restricted dialog width for better layout on wider screens

Hagellach37 reviewed (approved):
Nice!
